shared/src/main/java/com/couchbase/lite/AbstractDatabase.java

```diff
@@ -347,6 +347,8 @@ public void purge(String id) throws CouchbaseLiteException {
     public void setDocumentExpiration(String id, Date expiration) throws CouchbaseLiteException {
         synchronized (lock) {
             try {
+                if(getC4Database().get(id, true) == null)
+                    return;
```

Comment:
I think it should throw NotFound Exception.

Reply:
they are no longer there... I added them and removed them. The reason I removed them was because LiteCore will return NotFound exception when setting expiration on none existence document.

Reply:
I see.